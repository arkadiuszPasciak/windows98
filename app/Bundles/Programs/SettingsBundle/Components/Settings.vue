<template>
  <UIModal
    class="Settings"
    :title="t('Settings.title')"
    :resize-window="false"
    width="420px"
    height="415px"
    :modal-state="programStore.modalSettings"
    @close-modal="closeModal"
  >
    <UITabs :tabs="tabs" />
  </UIModal>
</template>

<script setup lang="ts">
import { useProgramStore } from "@APP|Bundles/ProgramBundle/Stores/Program.stores"
import SettingsDateTime from "@APP|Bundles/SettingsBundle/Components/SettingsDateTime.vue"
import SettingsGeneral from "@APP|Bundles/SettingsBundle/Components/SettingsGeneral.vue"
import SettingsLanguages from "@APP|Bundles/SettingsBundle/Components/SettingsLanguages.vue"
import UIModal from "@APP|Bundles/UIModalBundle/Components/UIModal.vue"
import UITabs from "@APP|Bundles/UITabsBundle/Components/UITabs.vue"
import type { TUITabsProps } from "@APP|Bundles/UITabsBundle/Supports/UITabs.supports"
import { useI18n } from "vue-i18n"

const { t } = useI18n()
const programStore = useProgramStore()

const tabs = [
	{
		title: "Settings.general",
		component: SettingsGeneral,
	},
	{
		title: "Settings.date-time",
		component: SettingsDateTime,
	},
	{
		title: "Settings.languages",
		component: SettingsLanguages,
	},
] as TUITabsProps

const closeModal = (): void => {
	programStore.updateSettingsModal(false)
}
</script>

<style lang="scss" scoped />
