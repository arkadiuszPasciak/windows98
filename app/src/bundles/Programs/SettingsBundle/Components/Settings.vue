<template>
  <UIModal
    class="Settings"
    :title="t('Settings.title')"
    :resize-window="false"
    width="420px"
    height="415px"
	:modal-state="programs.settings"
    @close-modal="closeModal"
  >
    <UITabs :tabs="tabs" />
  </UIModal>
</template>

<script setup lang="ts">
import type { TUITabsProps } from "@APP/src/bundles/UI/UITabsBundle/Supports/UITabs.supports"
import { useProgramManager } from "@APP/src/configs/app/hooks"
import SettingsDateTime from "@APP|Bundles/SettingsBundle/Components/SettingsDateTime.vue"
import SettingsGeneral from "@APP|Bundles/SettingsBundle/Components/SettingsGeneral.vue"
import SettingsLanguages from "@APP|Bundles/SettingsBundle/Components/SettingsLanguages.vue"
import UIModal from "@APP|Bundles/UIModalBundle/Components/UIModal.vue"
import UITabs from "@APP|Bundles/UITabsBundle/Components/UITabs.vue"
import { useI18n } from "vue-i18n"

const { t } = useI18n()
const { runProgram, programs } = useProgramManager()

const tabs = [
	{
		title: "Settings.general",
		component: SettingsGeneral,
	},
	{
		title: "Settings.date-time",
		component: SettingsDateTime,
	},
	{
		title: "Settings.languages",
		component: SettingsLanguages,
	},
] as TUITabsProps

const closeModal = (): void => {
	runProgram("settings", false)
}
</script>

<style lang="scss" scoped />
