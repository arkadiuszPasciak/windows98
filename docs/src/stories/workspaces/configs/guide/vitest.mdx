import { Meta, Title } from "@storybook/addon-docs/blocks";

<Meta title="Workspaces/Configs/Guide/Vitest" />

<Title>âš¡ Vitest</Title>

Complete guide to using Vitest configuration from `@windows98/configs` for unit and integration testing.

## Overview

Vitest configuration provides fast, modern testing setup optimized for TypeScript and modern JavaScript frameworks within the Windows98 ecosystem.

## Configuration Path

```
@windows98/configs/vitest/shared
```

### Install Dependencies

```bash
# Install vitest
pnpm add -D vitest

# Install the configs package
pnpm add -D @windows98/configs
```

## Basic Usage

### Simple Setup

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig(sharedConfig)
```

### With Custom Settings

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig({
  ...sharedConfig,
  test: {
    ...sharedConfig.test,
    environment: 'jsdom',
    setupFiles: ['./src/test-setup.ts']
  }
})
```

## Package.json Scripts

```json
{
  "scripts": {
	"test:unit:run": "vitest run -c ./@test/vitest/vitest.config.js",
	"test:unit:watch": "vitest -c ./@test/vitest/vitest.config.js",
  }
}
```

