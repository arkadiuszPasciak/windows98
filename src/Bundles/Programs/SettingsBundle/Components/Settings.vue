<template>
  <UIModal
    class="Settings"
    :title="t('Settings.title')"
    :resize-window="false"
    width="420px"
    height="415px"
    :modal-state="programStore.modalSettings"
    @close-modal="closeModal"
  >
    <UITabs :tabs="tabs" />
  </UIModal>
</template>

<script setup lang="ts">
  import { useI18n } from 'vue-i18n'
  import UIModal from '@Bundles/UIModalBundle/Components/UIModal.vue'
  import UITabs from '@Bundles/UITabsBundle/Components/UITabs.vue'
  import SettingsDateTime from '@Bundles/SettingsBundle/Components/SettingsDateTime.vue'
  import SettingsGeneral from '@Bundles/SettingsBundle/Components/SettingsGeneral.vue'
  import SettingsLanguages from '@Bundles/SettingsBundle/Components/SettingsLanguages.vue'
  import { useProgramStore } from '@Bundles/ProgramBundle/Stores/Program.stores'
  import { TUITabsProps } from '@Bundles/UITabsBundle/Supports/UITabs.supports'

  const { t } = useI18n()
  const programStore = useProgramStore()

  const tabs = [
    {
      title: 'Settings.general',
      component: SettingsGeneral,
    },
    {
      title: 'Settings.date-time',
      component: SettingsDateTime,
    },
    {
      title: 'Settings.languages',
      component: SettingsLanguages,
    },
  ] as TUITabsProps

  const closeModal = (): void => {
    programStore.updateSettingsModal(false)
  }
</script>

<i18n src="@Bundles/SettingsBundle/Locales/Settings.locales.json"></i18n>

<style lang="scss" scoped />
