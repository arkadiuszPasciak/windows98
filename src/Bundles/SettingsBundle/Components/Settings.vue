<template>
  <UIModal
    class="Settings"
    :title="t('Settings.title')"
    :resize-window="false"
    :width="335"
    :height="415"
    :modal-state="programStore.modalSettings"
    @close-modal="closeModal"
  >
    <UITabs :tabs="tabs" />
  </UIModal>
</template>

<script setup lang="ts">
  import { useI18n } from 'vue-i18n'
  import UIModal from '@Bundles/UIBundle/Components/UIModal.vue'
  import UITabs from '@Bundles/UIBundle/Components/UITabs.vue'
  import SettingsGeneral from '@Bundles/SettingsBundle/Components/SettingsGeneral.vue'
  import SettingsLanguages from '@Bundles/SettingsBundle/Components/SettingsLanguages.vue'
  import { useProgramStore } from '@Bundles/ProgramBundle/Stores/Program.stores'
  import { TUITabsProps } from '@Bundles/UIBundle/Supports/UITabs.supports'

  const { t } = useI18n()
  const programStore = useProgramStore()

  const tabs = [
    {
      title: 'Settings.general',
      component: SettingsGeneral,
    },
    {
      title: 'Settings.languages',
      component: SettingsLanguages,
    },
  ] as TUITabsProps

  const closeModal = (): void => {
    programStore.updateSettingsModal(false)
  }
</script>

<i18n src="@Bundles/SettingsBundle/Locales/Settings.locales.json"></i18n>

<style lang="scss" scoped></style>
