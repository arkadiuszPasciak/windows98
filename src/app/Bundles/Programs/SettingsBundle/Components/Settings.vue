<template>
  <UIModal
    class="Settings"
    :title="t('Settings.title')"
    :resize-window="false"
    width="420px"
    height="415px"
    :modal-state="programStore.modalSettings"
    @close-modal="closeModal"
  >
    <UITabs :tabs="tabs" />
  </UIModal>
</template>

<script setup lang="ts">
  import { useI18n } from 'vue-i18n'
  import UIModal from '@APP|Bundles/UIModalBundle/Components/UIModal.vue'
  import UITabs from '@APP|Bundles/UITabsBundle/Components/UITabs.vue'
  import SettingsDateTime from '@APP|Bundles/SettingsBundle/Components/SettingsDateTime.vue'
  import SettingsGeneral from '@APP|Bundles/SettingsBundle/Components/SettingsGeneral.vue'
  import SettingsLanguages from '@APP|Bundles/SettingsBundle/Components/SettingsLanguages.vue'
  import { useProgramStore } from '@APP|Bundles/ProgramBundle/Stores/Program.stores'
  import { TUITabsProps } from '@APP|Bundles/UITabsBundle/Supports/UITabs.supports'

  const { t } = useI18n()
  const programStore = useProgramStore()

  const tabs = [
    {
      title: 'Settings.general',
      component: SettingsGeneral,
    },
    {
      title: 'Settings.date-time',
      component: SettingsDateTime,
    },
    {
      title: 'Settings.languages',
      component: SettingsLanguages,
    },
  ] as TUITabsProps

  const closeModal = (): void => {
    programStore.updateSettingsModal(false)
  }
</script>

<i18n src="@APP|Bundles/SettingsBundle/Locales/Settings.locales.json"></i18n>

<style lang="scss" scoped />
