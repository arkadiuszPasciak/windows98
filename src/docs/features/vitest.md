# [Vitest](https://vitest.dev/)

## Description

Vitest is a modern and fast testing framework specifically designed for Vite-powered projects. Developed by the [Vite team](https://vitejs.dev/team.html), Vitest leverages Vite's architecture to provide a seamless testing experience with features like native ESM support, rapid test execution, and isolated testing environments. It offers compatibility with various front-end libraries and frameworks, such as Vue, React, and Svelte. With its straightforward configuration and integration with Vite, Vitest aims to enhance the developer experience by providing an efficient and enjoyable testing workflow for Vite-based web applications.

## Commands

[Available commands in the application](/reference/cli.html#vitest)
