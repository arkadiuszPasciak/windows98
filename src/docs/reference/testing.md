# Testing

This documentation provides an overview of the testing process employed in our application. We utilize two powerful testing tools to ensure the reliability and stability of the code: Cypress and Vitest. These tools are employed in different aspects of testing to maximize coverage and maintain code quality.

## [Vitest](https://vitest.dev/)

`@/vite.config.ts` (in the block test)

Vitest is a lightweight and fast testing framework specifically designed for Vite applications. It provides seamless integration with Vite, supporting both JavaScript and TypeScript code. Vitest is particularly well-suited for unit testing, which focuses on verifying the functionality of individual code units, such as functions, classes, or modules.

We use Vitest for `unit testing` our TypeScript code, ensuring that each function or class in our application performs as intended. This approach allows us to isolate potential issues and fix them before they cause problems in the broader application context.

To effectively use Vitest, we create a series of test suites that group related tests, with each test case focusing on a specific unit of code. These test cases cover various scenarios, such as input validation, error handling, and edge cases, to ensure comprehensive coverage of our codebase.
