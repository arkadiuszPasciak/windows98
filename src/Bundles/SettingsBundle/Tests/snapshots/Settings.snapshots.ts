import { mount } from '@vue/test-utils'
import { expect, it, describe } from 'vitest'
import { createTestingPinia } from '@pinia/testing'
import Settings from '@Bundles/SettingsBundle/Components/Settings.vue'

const defaultWrapper = mount(Settings, {
  global: {
    plugins: [
      createTestingPinia({
        initialState: {
          program: { modalSettings: true },
        },
      }),
    ],
  },
})

describe('[SettingsBundle]<Snapshots>(Settings)', async () => {
  it('renders correctly component', () => {
    expect(defaultWrapper.element).toMatchSnapshot()
  })
})
