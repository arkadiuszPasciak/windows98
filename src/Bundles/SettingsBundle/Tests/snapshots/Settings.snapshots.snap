// Vitest Snapshot v1

exports[`[SettingsBundle]<Snapshots>(Settings) > renders correctly component 1`] = `
<div
  class="UIModal move-window Settings"
  data-v-99a75817=""
>
  <div
    class="container"
    data-v-99a75817=""
    style="width: 335px; height: 415px;"
  >
    <div
      class="header"
      data-v-99a75817=""
      style="cursor: default;"
    >
      <h3
        class="title"
        data-v-99a75817=""
      >
        Settings
      </h3>
      <button
        class="UIButton size-small button-close"
        data-v-99a75817=""
        data-v-cf1bb3a9=""
        type="button"
      >
        
        <img
          alt="Close window"
          class="button-close-icon"
          data-v-99a75817=""
          height="7"
          src="/icons/close-modal.svg"
          width="8"
        />
        
        
        
        
        
      </button>
    </div>
    
    <div
      class="UITabs"
      data-v-39833990=""
    >
      <div
        class="tabs"
        data-v-39833990=""
      >
        
        <button
          class="tab active"
          data-v-39833990=""
          type="button"
        >
          General
        </button>
        <button
          class="tab"
          data-v-39833990=""
          type="button"
        >
          Date & Time
        </button>
        <button
          class="tab"
          data-v-39833990=""
          type="button"
        >
          Languages
        </button>
        
      </div>
      <div
        class="content"
        data-v-39833990=""
      >
        
        <div
          class="SettingsGeneral content-tab active"
          data-v-39833990=""
          data-v-40e21cf7=""
        >
          <img
            class="image"
            data-v-40e21cf7=""
            src="/icons/windows-settings.svg"
          />
          <h3
            class="title"
            data-v-40e21cf7=""
          >
            Device:
          </h3>
          <p
            class="text"
            data-v-40e21cf7=""
          >
            Desktop
          </p>
          <h3
            class="title"
            data-v-40e21cf7=""
          >
            Operation system:
          </h3>
          <p
            class="text"
            data-v-40e21cf7=""
          >
            undefined
          </p>
          <h3
            class="title"
            data-v-40e21cf7=""
          >
            Browser:
          </h3>
          <p
            class="text"
            data-v-40e21cf7=""
          >
            undefined
          </p>
        </div>
        <div
          class="SettingsDateTime content-tab"
          data-v-39833990=""
          data-v-b9c2b439=""
        >
          <select
            class="UISelect variant-undefined DateMonthSwitcher"
            data-v-4c307dbf=""
            data-v-b9c2b439=""
            data-v-d3c68ad1=""
            id="date-months"
            name="date-months"
          >
            
            <option
              class="option"
              data-v-d3c68ad1=""
              value="jan"
            >
              January
            </option>
            <option
              class="option"
              data-v-d3c68ad1=""
              value="feb"
            >
              February
            </option>
            <option
              class="option"
              data-v-d3c68ad1=""
              value="mar"
            >
              March
            </option>
            <option
              class="option"
              data-v-d3c68ad1=""
              value="apr"
            >
              April
            </option>
            <option
              class="option"
              data-v-d3c68ad1=""
              value="may"
            >
              May
            </option>
            <option
              class="option"
              data-v-d3c68ad1=""
              value="jun"
            >
              June
            </option>
            <option
              class="option"
              data-v-d3c68ad1=""
              value="jul"
            >
              July
            </option>
            <option
              class="option"
              data-v-d3c68ad1=""
              value="aug"
            >
              August
            </option>
            <option
              class="option"
              data-v-d3c68ad1=""
              value="sep"
            >
              September
            </option>
            <option
              class="option"
              data-v-d3c68ad1=""
              value="oct"
            >
              October
            </option>
            <option
              class="option"
              data-v-d3c68ad1=""
              value="nov"
            >
              November
            </option>
            <option
              class="option"
              data-v-d3c68ad1=""
              value="dec"
            >
              December
            </option>
            
          </select>
          <div
            class="UIInput DateYearSwitcher"
            data-v-04f77c22=""
            data-v-b9c2b439=""
            data-v-e394574e=""
          >
            <!--v-if-->
            <input
              class="input"
              data-v-e394574e=""
              id="year-switcher"
              max="2050"
              min="1950"
              type="number"
            />
          </div>
          <div
            class="DateCalendar"
            data-v-4e49242e=""
            data-v-b9c2b439=""
          >
            <div
              class="heading"
              data-v-4e49242e=""
            >
              
              <div
                class="heading-item"
                data-v-4e49242e=""
              >
                M
              </div>
              <div
                class="heading-item"
                data-v-4e49242e=""
              >
                T
              </div>
              <div
                class="heading-item"
                data-v-4e49242e=""
              >
                W
              </div>
              <div
                class="heading-item"
                data-v-4e49242e=""
              >
                T
              </div>
              <div
                class="heading-item"
                data-v-4e49242e=""
              >
                F
              </div>
              <div
                class="heading-item"
                data-v-4e49242e=""
              >
                S
              </div>
              <div
                class="heading-item"
                data-v-4e49242e=""
              >
                S
              </div>
              
            </div>
            <div
              class="content"
              data-v-4e49242e=""
            >
              
              <button
                class="content-item"
                data-v-4e49242e=""
                type="button"
              >
                1
              </button>
              <button
                class="content-item"
                data-v-4e49242e=""
                type="button"
              >
                2
              </button>
              <button
                class="content-item"
                data-v-4e49242e=""
                type="button"
              >
                3
              </button>
              <button
                class="content-item"
                data-v-4e49242e=""
                type="button"
              >
                4
              </button>
              <button
                class="content-item"
                data-v-4e49242e=""
                type="button"
              >
                5
              </button>
              <button
                class="content-item"
                data-v-4e49242e=""
                type="button"
              >
                6
              </button>
              <button
                class="content-item"
                data-v-4e49242e=""
                type="button"
              >
                7
              </button>
              <button
                class="content-item"
                data-v-4e49242e=""
                type="button"
              >
                8
              </button>
              <button
                class="content-item"
                data-v-4e49242e=""
                type="button"
              >
                9
              </button>
              <button
                class="content-item"
                data-v-4e49242e=""
                type="button"
              >
                10
              </button>
              <button
                class="content-item"
                data-v-4e49242e=""
                type="button"
              >
                11
              </button>
              <button
                class="content-item"
                data-v-4e49242e=""
                type="button"
              >
                12
              </button>
              <button
                class="content-item"
                data-v-4e49242e=""
                type="button"
              >
                13
              </button>
              <button
                class="content-item"
                data-v-4e49242e=""
                type="button"
              >
                14
              </button>
              <button
                class="content-item"
                data-v-4e49242e=""
                type="button"
              >
                15
              </button>
              <button
                class="content-item"
                data-v-4e49242e=""
                type="button"
              >
                16
              </button>
              <button
                class="content-item"
                data-v-4e49242e=""
                type="button"
              >
                17
              </button>
              <button
                class="content-item"
                data-v-4e49242e=""
                type="button"
              >
                18
              </button>
              <button
                class="content-item"
                data-v-4e49242e=""
                type="button"
              >
                19
              </button>
              <button
                class="content-item"
                data-v-4e49242e=""
                type="button"
              >
                20
              </button>
              <button
                class="content-item"
                data-v-4e49242e=""
                type="button"
              >
                21
              </button>
              <button
                class="content-item"
                data-v-4e49242e=""
                type="button"
              >
                22
              </button>
              <button
                class="content-item"
                data-v-4e49242e=""
                type="button"
              >
                23
              </button>
              <button
                class="content-item"
                data-v-4e49242e=""
                type="button"
              >
                24
              </button>
              <button
                class="content-item"
                data-v-4e49242e=""
                type="button"
              >
                25
              </button>
              <button
                class="content-item"
                data-v-4e49242e=""
                type="button"
              >
                26
              </button>
              <button
                class="content-item"
                data-v-4e49242e=""
                type="button"
              >
                27
              </button>
              <button
                class="content-item"
                data-v-4e49242e=""
                type="button"
              >
                28
              </button>
              <button
                class="content-item"
                data-v-4e49242e=""
                type="button"
              >
                29
              </button>
              <button
                class="content-item"
                data-v-4e49242e=""
                type="button"
              >
                30
              </button>
              
            </div>
          </div>
        </div>
        <div
          class="SettingsLanguages content-tab"
          data-v-39833990=""
        >
          <select
            class="UISelect variant-undefined LanguageSwitcher"
            data-v-56cc1592=""
            data-v-d3c68ad1=""
            id="languages"
            name="languages"
          >
            
            <option
              class="option"
              data-v-d3c68ad1=""
              value="en"
            >
              English
            </option>
            <option
              class="option"
              data-v-d3c68ad1=""
              value="pl"
            >
              Polish
            </option>
            
          </select>
          <select
            class="UISelect variant-undefined ThemeSwitcher"
            data-v-6a01b610=""
            data-v-d3c68ad1=""
            id="themes"
            name="themes"
          >
            
            <option
              class="option"
              data-v-d3c68ad1=""
              value="light"
            >
              Light
            </option>
            <option
              class="option"
              data-v-d3c68ad1=""
              value="dark"
            >
              Dark
            </option>
            
          </select>
        </div>
        
      </div>
    </div>
    
  </div>
</div>
`;
